### PR TITLE
Refactor to gradle kts, and improve clarity of modules

### PR DESCRIPTION
**Pull request by vLuckyyy**
Refactor project build:

- Upgrade Gradle version to 8.0.2 for better performance and stability.
- Enhance module clarity by renames module
- Introduce shadowJar task to produce a single executable jar with all dependencies included.
- Improve overall build efficiency and simplify future maintenance.

### CODE REVIEW

File: bedwars-plugin/src/main/resources/plugin.yml
Comment (tomkeuper) [inline bedwars-plugin/src/main/resources/plugin.yml:1]: Did you delete it on purpose? I cant find it anywhere else

Comment (vLuckyyy) [reply bedwars-plugin/src/main/resources/plugin.yml:1]: Yes, but i replaced with gradle plugin for this:
https://github.com/vLuckyyy/BedWars1058/blob/cd1ef9df056fc7d17692506be2a4d55f2db061ab/plugin/build.gradle.kts#L50

**Comment by tomkeuper** [comment]
Just out out curiosity; In the last PR people said, one of the main advantages of gradle was speed. How long does it take gradle to build and package all bedwars projects?

**Comment by vLuckyyy** [comment]
> Just out out curiosity; In the last PR people said, one of the main advantages of gradle was speed. How long does it take gradle to build and package all bedwars projects?

It's important to note that the build time for Gradle can vary depending on several factors, such as the system configuration of Gradle, the version of Gradle being used (in this case, 8.0.2), and the computer's specifications. It's also worth mentioning that each subsequent build will generally be faster than the previous one, as the initial build involves caching.

Regarding the full build time for the final Jar project, using the shadowJar task without any changes immediately after indexing the project and removing any previously stored build files and cache folders, here are some examples:
1st attempt - 24s
2nd attempt - 2s
3rd attempt - 2s

After making a one-line change, the build time is as follows:
1st attempt - 24s
2nd attempt - 4s
3rd attempt - 1s

There is also an option in the compiler to compile only the classes that have been changed (isIncremental). After configuring it, the compilation times for making a one-line change are:
1st attempt - 9s
2nd attempt - 2s
3rd attempt - 1s

Additionally, I've included my Gradle configuration in the gradle.properties file located at `C:\Users\USER\.gradle\gradle.properties`:

```properties
# Gradle Daemon Configuration
org.gradle.daemon=true
org.gradle.jvmargs=-Xmx4096M -XX:MaxMetaspaceSize=2048m -XX:+HeapDumpOnOutOfMemoryError

# Parallel Build Configuration
org.gradle.parallel=true
org.gradle.workers.max=8
org.gradle.parallel.intra=true

# Gradle Cache Configuration
org.gradle.caching=true
org.gradle.build-cache.local.only=true

# Task Execution Configuration
org.gradle.configureondemand=true
org.gradle.unsafe.watch-fs=true
org.gradle.vfs.file-mode-watching=false
```

**Pull request closed without merging**